### PR TITLE
Carousel TS Convert

### PR DESCRIPTION
**Pull request by jaketrent**
yeeehaa!

### CODE REVIEW

File: packages/carousel/src/js/__specs__/utils.spec.ts [hunk 1]
```diff
@@ -56,43 +56,6 @@ describe('utils', () => {
     })
   })
 
-  describe('#combineFns', () => {
```
Comment (jaketrent) [inline packages/carousel/src/js/__specs__/utils.spec.ts:59]: using util pkg instead

File: packages/carousel/src/react/__specs__/index.spec.tsx [hunk 2]
```diff
@@ -30,16 +30,22 @@ describe('Carousel', () => {
...
     it('should hide the prev control', () => {
-      const control = container
-        .querySelector('[aria-label="get previous carousel page"]')
-        .closest('[data-testid="carousel control"]')
+      const page = container.querySelector(
+        '[aria-label="get previous carousel page"]'
+      )
+      expect(page).toBeDefined()
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      const control = page!.closest('[data-testid="carousel control"]')
```
Comment (jaketrent) [inline packages/carousel/src/react/__specs__/index.spec.tsx:38]: I like the assertion here.  I don't like that I have to disable eslint.  The alternative is to put an `if` block around this.

File: packages/carousel/src/react/context.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  offset: number
+  prev: () => void
+  id: string
+  pageCount: number
+  perPage: number
+  transitioning: boolean
+  setTransitioning: React.Dispatch<React.SetStateAction<boolean>>
+}
+const context = React.createContext<CarouselContextValue>({
+  activePage: 0,
```
Comment (jaketrent) [inline packages/carousel/src/react/context.ts:15]: Some of these defaults are new.  Simplified the typing process. Double-check my logic that having these are ok as defaults.

File: packages/carousel/src/react/__specs__/index.spec.tsx (hunk 2 above)
Comment (danethurber) [reply packages/carousel/src/react/__specs__/index.spec.tsx:38]: maybe disable this rule for .spec files 🤷

File: packages/carousel/src/react/control.tsx [hunk 4]
```diff
@@ -34,33 +34,49 @@ const styles = {
...
 
-export function Control(props) {
+interface ControlProps
+  extends Omit<React.HTMLAttributes<HTMLButtonElement>, 'onClick'> {
+  direction: ValueOf<typeof vars.controlDirections>
+  onClick?: (evt?: React.MouseEvent) => void
+}
+interface ControlStatics {
+  directions: typeof vars.controlDirections
+}
```
Comment (danethurber) [inline packages/carousel/src/react/control.tsx:44]: did you need to replace the `onClick` fn. seems like the types are the same

File: packages/carousel/src/react/index.tsx [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+                key={pageIndex}
+                isActivePage={isActivePage}
+                paged={pager.paged}
+              >
+                {items.map(
+                  (item: React.ReactElement | null, itemIndex: number) => {
+                    const wrapped = !!item && isOfComponentType(item, Item)
+                    if (!wrapped) item = <Item key={itemIndex}>{item}</Item>
+
+                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```
Comment (danethurber) [inline packages/carousel/src/react/index.tsx:114]: one way to do this is to `{React.isActiveElement(item) && ...}`

File: packages/carousel/src/react/use-swipe.ts [hunk 6]
```diff
@@ -23,10 +37,10 @@ export default function useSwipe(ref, opts = {}) {
     timeThreshold
   } = { ...defaultOpts, ...opts }
 
-  const timeout = useRef()
+  const timeout = useRef<NodeJS.Timeout>()
 
-  const [swipping, setSwipping] = useState(false)
```
Comment (danethurber) [inline packages/carousel/src/react/use-swipe.ts:28]: 🤦

**Review by danethurber — APPROVED** [review]
awesome work!

File: packages/carousel/src/react/control.tsx (hunk 4 above)
Comment (jaketrent) [reply packages/carousel/src/react/control.tsx:44]: yeah, great point.  Not sure why I did that. Probably just a too-literal conversion from proptypes.

File: packages/carousel/src/react/index.tsx (hunk 5 above)
Comment (jaketrent) [reply packages/carousel/src/react/index.tsx:114]: seems like I tried that option.  Not sure what the outcome was or why I chose this one.  There's a weekend in between in my mind.  

Is there an approach you think we should prefer?

File: packages/carousel/src/react/control.tsx (hunk 4 above)
Comment (jaketrent) [reply packages/carousel/src/react/control.tsx:44]: Well, doesn't make sense in isolation why I defined line 40.  But when I remove, I get down to combineFns, and I can't harangue Params to be typed correctly.

**Pull request merged**